ehrql/tables/tpp.py

```diff
@@ -552,6 +553,157 @@ class clinical_events_ranges(EventFrame):
+    may or may not have actually received the treatment or completed the course (but we
+    assume that they usually do). They may have another form completed for another
+    treatment, either because it was decided to give them a different treatment, or for
+    some other reason. They may in theory also have another form completed some months
+    later for another instance of infection.
+
+    Treatment dates may be in the past or future at the point when the form is
+    submitted.
+
+    Note that this dataset contains some **duplicate** rows – some full duplicates and
+    some partial duplicates.
+
+
+    **More Information**
+
+    * [Treatment guidelines](https://www.nice.org.uk/guidance/ta878)
+    * [Draft Data Report](https://docs.google.com/document/d/15o4x9sqHEO-sLm2dTqgm3PyAh72cdgOOmZC4AB3BTNk/) (currently only available to internal staff)
+    """
+
+    age_at_received_date = Series(
+        int,
+        description="""
+            Can occasionally be zero, presumably indicating an unknown or missing value
+            as minimum eligibility age is 12.
+        """,
+    )
+    count = Series(
+        int,
+        description="Number of forms.",
+    )
+    covid_indication = Series(
+        str,
+        description="Treatment setting/indication.",
+        constraints=[
+            Constraint.Categorical(
+                ["non_hospitalised", "hospitalised_with", "hospital_onset"]
+            )
+        ],
+    )
+    current_status = Series(
+        str,
+        description="Status of form/application.",
+        constraints=[
+            Constraint.Categorical(
+                [
+                    "Approved",
+                    "Treatment Complete",
+                    "Treatment Not Started",
+                    "Treatment Stopped",
+                ]
+            )
+        ],
+    )
+    diagnosis = Series(
+        str,
+        description="Always has the value 'Covid-19'.",
+    )
+    form_name = Series(
+        str,
+        description="""
+            Name and version of the patient registration form used to register the
+            treatment.
+        """,
+    )
+    intervention = Series(
+        str,
+        description="""
+            Intervention or therapeutic name. Expected to be one of:
+
+             * Paxlovid
+             * Sotrovimab
+             * Molnupiravir
+             * Remdesivir
+             * Casirivimab and imdevimab
+        """,
+    )
+    load_date = Series(
+        datetime.date,
+        description="Date on which the current dataset was imported.",
+    )
+    received = Series(
+        datetime.date,
+        description="Date form submitted.",
+    )
+    region = Series(
+        str,
+        description="""
+            NHS England region in which the CMDU submitting the form is located.
+        """,
+    )
+    risk_cohort = Series(
+        str,
+        description="""
```

Comment:
I think this description needs to refer to the fact that it's a combined field of comma-separated values and comes from one of the 3 intervention risk groups (also possibly the fact that the data doesn't include risk groups for Paxlovid and Remdesivir - need to check with Millie). Also we should double-check the list of diseases (might be able to just do a release from Airlock for the therapeutics-risk-conditions workspace).

Reply:
oh, right, the ticket said 

> Note that we don't need to worry about duplicate risk groups across the 3 risk group columns because only one of those contains data in any one row

so my expectation was that this would never return comma-separated values to the researcher (all commas would be introduced & also removed during the cleaning) - but I haven't verified this for real, maybe we should warn researchers here?

Reply:
oh, I see, it's only of these named fields, but the field can contain multiple groups. So the previous docs were inaccurate?

Reply:
So a single risk group column can have multiple values in it - they're joined with "ands" in the raw values, like "Liver disease and cancer". The cleaning on the 3 separate columns is removing those and joining with commas (with all the REPLACE and COALESCE and STUFFing), so the value in that column becomes "Liver disease, cancer". In cohort-extractor it does that part when it creates the temporary table, and then it selects from that and joins across the 3 columns, but it'll still need to clean up the values (or make the combined column in a subsequent query). 
(I think the tests might make this clearer)